src/vs/code/electron-main/lifecycle.ts

```diff
@@ -5,19 +5,18 @@
 
 'use strict';
 
-import Uri from 'vs/base/common/uri';
 import { EventEmitter } from 'events';
 import { ipcMain as ipc, app } from 'electron';
 import { TPromise, TValueCallback } from 'vs/base/common/winjs.base';
 import { ReadyState, VSCodeWindow } from 'vs/code/electron-main/window';
 import { IEnvironmentService } from 'vs/platform/environment/common/environment';
 import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
-import { IBackupMainService } from 'vs/platform/backup/common/backup';
 import { ILogService } from 'vs/code/electron-main/log';
 import { IStorageService } from 'vs/code/electron-main/storage';
 
 const EventTypes = {
-	BEFORE_QUIT: 'before-quit'
+	BEFORE_QUIT: 'before-quit',
+	AFTER_UNLOAD: 'after-unload'
```

Comment:
@Tyriar should this maybe be `BEFORE_CLOSE`? because the next thing that is going to happen after this event is the window closing.